ElectronClient/app/gui/ResourceScreen.tsx

```diff
@@ -0,0 +1,180 @@
+import * as React from 'react';
+import { groupBy, orderBy } from 'lodash';
+
+const { connect } = require('react-redux');
+const { _ } = require('lib/locale.js');
+const { themeStyle } = require('../theme.js');
+const { bridge } = require('electron').remote.require('./bridge');
+const { Header } = require('./Header.min.js');
+const prettyBytes = require('pretty-bytes');
+const Resource = require('lib/models/Resource.js');
+const NoteResource = require('lib/models/NoteResource.js');
+
+interface IProps {
+	style: any;
+	theme: any;
+}
+
+interface IResource {
+	title: string
+	id: string
+	isOrphan: boolean
+	size: string
+	daoResource: any
+}
+
+interface IState {
+	resources: IResource[] | undefined
+	sortingOrders: SortingOrder[]
+}
+
+interface IResourceTable {
+	resources: IResource[]
+	sortingOrders: SortingOrder[]
+	onResourceClick: (resource: IResource) => any
+	onResourceDelete: (resource: IResource) => any
+	onToggleSortOrder: (sortOrder: SortingOrder) => any
+}
+
+type SortingOrder = 'size' | 'name' | 'orphaned'
+
+const sortIteratees: { [order in SortingOrder]: (resource: IResource) => any } = {
+	orphaned: (resource: IResource) =>
+		resource.isOrphan ? 0 : 1,
+	size: (resource: IResource) =>
+		1 / resource.daoResource.size,
+	name: (resource: IResource) =>
+		resource.title,
+};
+
+const getIteratees = (sortingOrders: SortingOrder[]) => {
+	const ordering = (Object.keys(sortIteratees) as SortingOrder[])
+		.filter(order => sortingOrders.includes(order));
+	return ordering.map(order => sortIteratees[order]);
+};
+
+const isOrphan = (noteResources: any[] | undefined): boolean => {
+	if (!noteResources) {
+		return false;
+	}
+	return noteResources.filter((nr) => nr.is_associated === 1).length === 0;
+};
+
+const buildResources = (resources: any[], noteResources: any[]) => {
+	const groupedNoteResources = groupBy(noteResources, 'resource_id');
+	return resources.map((resource: any) => ({
+		title: resource.title,
+		id: resource.id,
+		isOrphan: isOrphan(groupedNoteResources[resource.id]),
+		daoResource: resource,
+		size: prettyBytes(resource.size),
+	} as IResource));
```

Comment:
Likewise, if you are going to load all the resources, then perhaps you shouldn't also filter them all like this, as that might be slow. Data formatting ideally should be done when rendering the table but internally we should just keep the regular resource properties.